test/Sets/SimpleSparsePolynomialZonotope.jl

```diff
@@ -67,11 +67,19 @@ for N in [Float64, Float32, Rational{Int}]
     q1 = quadratic_map(Q, S2)
 
     @test center(q1) ≈ [-0.24, -0.416]
-    @test genmat(q1) ≈ [-1.2   0.16    0.0  0.4   0.4   0.0;
-                         -2.72  0.192  -3.2  0.48  0.48  0.0]
+    @test genmat(q1) ≈ [-1.2   0.16    0.0  0.8;
+                         -2.72  0.192  -3.2  0.96]
 
-    @test expmat(q1) == [ 1  0  2  1  1  0;
-                          0  1  0  1  1  2]
+    @test expmat(q1) == [ 1  0  2  1;
+                          0  1  0  1]
+
+    q2 = quadratic_map(Q, S2, S2)
+    @test center(q2) ≈ [-0.24, -0.416]
+    @test genmat(q2) ≈ [-1.2   0.16    0.0  0.8;
+                         -2.72  0.192  -3.2  0.96]
+
+    @test expmat(q2) == [ 1  0  2  1;
+                          0  1  0  1]
```

Comment:
If the tests are only written for `Float64`, they should be removed from the `for N in [Float64, Float32, Rational{Int}]` loop because that would just run the exact same tests three times. You can add a loop `for N in [Float64]` below and move the code there (the purpose of this loop is to avoid warnings about local/global variables from other test files).

Reply:
the SSPZ is created in different numerical precisions 

```julia
S2 = SimpleSparsePolynomialZonotope(N[0.2, -0.6], N[1 0;0 0.4], [1 0;0 1])
```

however the vector `Q` has only Float64 atm.

Changing the precision of the tests on the right hand side of the test (e.g. `@test center(q2) ≈ [-0.24, -0.416]`) fixes the tests, but maybe if feels a little like cheating, because the lhs of the test is always float64?) Changing the precision of also `Q` causes overflow error with `Rational{Int}`

Reply:
suggestions / comments on what to do?

Reply:
I will play around with it and send a commit.

Reply:
Done. The problem is that if you create an array with `3.2`, this will be a `Float64` and the `N` conversion happens only later. But `Float64` generally does not give you a "nice" rational:

```julia
julia> Rational{Int}(3.2)
3602879701896397//1125899906842624
```

So instead you need to create the numbers as `Rational`s and then convert to `N`.